### PR TITLE
Add listener+cb as first version of consumer SDK

### PR DESCRIPTION
**Pull request by JaylinYu**
support search and return msg by single key for now

### CODE REVIEW

**Review by RanMaoyi — APPROVED** [review]

**Comment by RanMaoyi** [comment]
Cool

File: src/mqtt/protocol/exchange/exchange_server.c [hunk 1]
```diff
@@ -438,6 +470,178 @@ exchange_client_get_msgs_by_key(void *arg, uint64_t key, uint32_t count, nng_msg
...
+	msg = nni_aio_get_msg(&p->ex_aio);
+	nni_aio_set_msg(&p->ex_aio, NULL);
+	nni_msg_set_pipe(msg, nni_pipe_id(p->pipe));
+
+	body    = nni_msg_body(msg);
+	len     = nni_msg_len(msg);
+
+	nni_mtx_lock(&sock->mtx);
+	// process query
+	char *keystr = (char *) (body + 4);
```
Comment (wanghaEMQ) [inline src/mqtt/protocol/exchange/exchange_server.c:506]: What's your idea about the format of payload in msg

**Review by wanghaEMQ — COMMENTED** [review]
It will not change the logic of origin exchange, right?

File: src/mqtt/protocol/exchange/exchange_server.c (hunk 1 above)
Comment (JaylinYu) [reply src/mqtt/protocol/exchange/exchange_server.c:506]: TBD

**Pull request merged**